core/src/toga/widgets/button.py

```diff
@@ -48,7 +54,17 @@ def __init__(
         # Set a dummy handler before installing the actual on_press, because we do not want
         # on_press triggered by the initial value being set
         self.on_press = None
-        self.text = text
+
+        # Set the content of the button - either an icon, or text, but not both.
+        if text:
```

Comment:
Good point - although more generally, this gives a valid path for `toga.Button()` to work, because `None` is a valid value for both `text` and `icon`.